### PR TITLE
AM-2405

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-2405

### Change description ###
Add drools rule config and tests for new case manager global roles

### CODE REVIEW

File: src/main/resources/validationrules/core/case-manager-global.drl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.GrantType;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleType
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+
+/*
+* All services: "admin-case-manager" and "ctsc-case-manager" case roles
+*/
+rule "create_case_manager_role"
+when
+    $rq: Request(assignerId == authenticatedUserId)
```
Comment (mattnayler) [inline src/main/resources/validationrules/core/case-manager-global.drl:14]: Can you add a FeatureFlag to the rule so the code can be released behind one.

File: src/main/resources/validationrules/core/case-manager-global.drl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    update($ra);
+    logMsg("Rule : create_case_manager_role");
+end;
+
+/*
+* All services: "admin-case-manager" and "ctsc-case-manager" case roles deletion
+*/
+rule "delete_case_manager_role"
+when
+   $rq: Request(assignerId == authenticatedUserId)
```
Comment (mattnayler) [inline src/main/resources/validationrules/core/case-manager-global.drl:35]: Can you add a FeatureFlag to the rule so the code can be released behind one.

File: src/main/resources/validationrules/core/case-manager-global.drl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleType
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+
+/*
+* All services: "admin-case-manager" and "ctsc-case-manager" case roles
+*/
+rule "create_case_manager_role"
+when
+    $rq: Request(assignerId == authenticatedUserId)
+    $ra: RoleAssignment(
```
Comment (mattnayler) [inline src/main/resources/validationrules/core/case-manager-global.drl:15]: Should this filter use the `CaseAllocatorApproval` in a similar way as the IAC `lead-judge` does etc see  https://github.com/hmcts/am-role-assignment-service/blob/master/src/main/resources/validationrules/iac/iac-case-role-validation-judicial.drl#L28

File: src/main/resources/validationrules/core/case-manager-global.drl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    logMsg("Rule : create_case_manager_role");
+end;
+
+/*
+* All services: "admin-case-manager" and "ctsc-case-manager" case roles deletion
+*/
+rule "delete_case_manager_role"
+when
+   $rq: Request(assignerId == authenticatedUserId)
+   $ra: RoleAssignment(
```
Comment (mattnayler) [inline src/main/resources/validationrules/core/case-manager-global.drl:36]: Should this filter use the `CaseAllocatorApproval` in a similar way as the IAC `lead-judge` does etc see  https://github.com/hmcts/am-role-assignment-service/blob/master/src/main/resources/validationrules/iac/iac-case-role-validation-judicial.drl#L153

**Comment by tomxelliott** [comment]
Closing as ticket has been moved back into the backlog pending services requesting the roles.

**Pull request closed without merging**